src/extractor/edge_based_graph_factory.cpp

```diff
@@ -638,6 +595,474 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
+                                     turn_candidates[getLeft(getLeft(conflict_end))].instruction,
+                                     RESOLVE_TO_LEFT))
+                            continue;
+                    }
+                }
+                else if (conflict_size >= 3)
+                {
+                    // a conflict of size larger than three cannot be handled with the current
+                    // model.
+                    // Handle it as best as possible and keep the rest of the conflicting turns
+                    if (conflict_size > 3)
+                    {
+                        NodeID conflict_location = m_node_based_graph->GetTarget(via_eid);
+                        util::SimpleLogger().Write()
+                            << "[warning] found conflict larget than size three at "
+                            << m_node_info_list[conflict_location].lat << ", "
+                            << m_node_info_list[conflict_location].lon << std::endl;
+                    }
+
+                    if (!resolve(candidate_at_begin.instruction, instruction_right_of_begin,
+                                 RESOLVE_TO_RIGHT))
+                    {
+                        if (isSlightTurn(turn.instruction))
+                            resolveTransitive(
+                                candidate_at_begin.instruction, instruction_right_of_begin,
+                                turn_candidates[getRight(getRight(conflict_begin))].instruction,
+                                RESOLVE_TO_RIGHT);
+                        else if (isSharpTurn(turn.instruction))
+                            resolveTransitive(
+                                candidate_at_end.instruction, instruction_left_of_end,
+                                turn_candidates[getLeft(getLeft(conflict_end))].instruction,
+                                RESOLVE_TO_LEFT);
+                    }
+                    if (!resolve(candidate_at_end.instruction, instruction_left_of_end,
+                                 RESOLVE_TO_LEFT))
+                    {
+                        if (isSlightTurn(turn.instruction))
+                            resolveTransitive(
+                                candidate_at_end.instruction, instruction_left_of_end,
+                                turn_candidates[getLeft(getLeft(conflict_end))].instruction,
+                                RESOLVE_TO_LEFT);
+                        else if (isSharpTurn(turn.instruction))
+                            resolveTransitive(
+                                candidate_at_begin.instruction, instruction_right_of_begin,
+                                turn_candidates[getRight(getRight(conflict_begin))].instruction,
+                                RESOLVE_TO_RIGHT);
+                    }
+                }
+            }
+        }
+    }
+    return turn_candidates;
+}
+
+bool EdgeBasedGraphFactory::isObviousChoice(EdgeID via_eid,
```

Comment:
@daniel-j-h did some updates. Could you review this function and suppressTurns again. After that we should be mostly done for now.